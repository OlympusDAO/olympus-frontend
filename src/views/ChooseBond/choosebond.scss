#choose-bond-view {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  overflow: hidden;
  .ohm-card .bond-hero h4 {
    font-weight: 600 !important;

    .MuiSkeleton-root {
      margin: auto;
    }
  }

  .claim-bonds-card {
    .global-claim-buttons {
      width: 100%;
      margin: 15px 0px;
      button {
        max-width: 233px;
      }
    }
    .small.global-claim-buttons {
      flex-direction: column;
      justify-content: space-around;
      align-items: center;
      flex-wrap: wrap;
      button {
        max-width: unset;
        width: 90%;
      }
    }
  }

  .bond-data-card {
    margin: auto;
    .ohm-pairs {
      display: flex;
      margin-right: 15px;
    }
    .bond-pair {
      display: flex !important;
      align-items: center;
      justify-content: left;
      margin-bottom: 15px;
      .MuiBox-root {
        margin-right: 10px;
      }
      .bond-name {
        svg {
          margin-left: 5px;
          vertical-align: middle;
          font-size: 15px;
        }
      }
    }
    .data-row {
      margin-bottom: 13px;

      .bond-price {
        display: flex;
        align-items: center;

        svg {
          margin-right: 5px;
        }
      }
    }
    .bond-link {
      margin-top: 10px;
      text-align: center;
      a {
        width: 96%;
      }
    }
  }

  .currency-icon {
    display: inline-block;
    min-width: 15px;
  }

  .MuiTable-root {
    margin: 10px 0px;
  }

  .claim-bond-card-buttons {
    flex-wrap: wrap;
    width: 100%;
    button {
      min-width: 48%;
      margin: 5px 0px;
    }
  }

  .MuiTableCell-root {
    p {
      margin: 0 !important;
      display: flex !important;
      align-items: center;
      // justify-content: flex-start;
      white-space: nowrap;
      img {
        margin-right: 3px;
      }
    }
  }

  .MuiTableBody-root .MuiTableRow-root .MuiTableCell-alignLeft.bond-name-cell {
    justify-content: space-between;
    display: flex;
    div.bond-name {
      width: 60%;
      margin-left: 5px !important;
      display: flex;
      flex-direction: column;
      justify-content: center;
      text-align: left;
      align-items: flex-start;
      p {
        margin-bottom: unset !important;
        line-height: 140% !important;
      }
      svg {
        margin-left: 3px !important;
        font-size: 1.2em !important;
      }
    }
  }
}

.tablet #choose-bond-view {
  .claim-bond-card-buttons {
    button {
      width: 40%;
    }
  }
}

.mobile #choose-bond-view {
  .claim-bond-card-buttons {
    flex-direction: column;
    button {
      width: 100%;
    }
  }
}
