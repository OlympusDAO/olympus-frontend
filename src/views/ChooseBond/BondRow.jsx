import { trim } from "../../helpers";
import BondLogo from "../../components/BondLogo";
import { Box, Button, Link, Paper, Typography, TableRow, TableCell, SvgIcon, Slide } from "@material-ui/core";
import { ReactComponent as ArrowUp } from "../../assets/icons/arrow-up.svg";
import { NavLink } from "react-router-dom";
import "./choosebond.scss";
import { Skeleton } from "@material-ui/lab";
import useBonds from "src/hooks/Bonds";

export function BondDataCard({ bond }) {
  const { loading } = useBonds();
  const isBondLoading = !bond.bondPrice ?? true;

  return (
    <Slide direction="up" in={true}>
      <Paper id={`${bond.name}--bond`} className="bond-data-card ohm-card">
        <div className="bond-pair">
          <BondLogo bond={bond} />
          <div className="bond-name">
            <Typography>{bond.displayName}</Typography>
            {bond.isLP && (
              <div>
                <Link href={bond.lpUrl} target="_blank">
                  <Typography variant="body1">
                    View Contract
                    <SvgIcon component={ArrowUp} htmlColor="#A3A3A3" />
                  </Typography>
                </Link>
              </div>
            )}
          </div>
        </div>

        <div className="data-row">
          <Typography>Price</Typography>
          <Typography className="bond-price">
            <>{isBondLoading ? <Skeleton width="50px" /> : trim(bond.bondPrice, 2)}</>
          </Typography>
        </div>

        <div className="data-row">
          <Typography>ROI</Typography>
          <Typography>{isBondLoading ? <Skeleton width="50px" /> : `${trim(bond.bondDiscount * 100, 2)}%`}</Typography>
        </div>

        <div className="data-row">
          <Typography>Purchased</Typography>
          <Typography>
            {isBondLoading ? (
              <Skeleton width="80px" />
            ) : (
              new Intl.NumberFormat("en-US", {
                style: "currency",
                currency: "USD",
                maximumFractionDigits: 0,
                minimumFractionDigits: 0,
              }).format(bond.purchased)
            )}
          </Typography>
        </div>
        <Link component={NavLink} to={`/bonds/${bond.name}`}>
          <Button variant="outlined" color="primary" fullWidth>
            <Typography variant="h5">Bond {bond.displayName}</Typography>
          </Button>
        </Link>
      </Paper>
    </Slide>
  );
}

export function BondTableData({ bond }) {
  // Use BondPrice as indicator of loading.
  const isBondLoading = !bond.bondPrice ?? true;
  // const isBondLoading = useSelector(state => !state.bonding[bond]?.bondPrice ?? true);

  return (
    <TableRow id={`${bond.name}--bond`}>
      <TableCell align="left" className="bond-name-cell">
        <BondLogo bond={bond} />
        <div className="bond-name">
          <Typography variant="body1">{bond.displayName}</Typography>
          {bond.isLP && (
            <Link color="primary" href={bond.lpUrl} target="_blank">
              <Typography variant="body1">
                View Contract
                <SvgIcon component={ArrowUp} htmlColor="#A3A3A3" />
              </Typography>
            </Link>
          )}
        </div>
      </TableCell>
      <TableCell align="left">
        <Typography>
          <>
            <span className="currency-icon">$</span>
            {isBondLoading ? <Skeleton width="50px" /> : trim(bond.bondPrice, 2)}
          </>
        </Typography>
      </TableCell>
      <TableCell align="left">{isBondLoading ? <Skeleton /> : `${trim(bond.bondDiscount * 100, 2)}%`}</TableCell>
      <TableCell align="right">
        {isBondLoading ? (
          <Skeleton />
        ) : (
          new Intl.NumberFormat("en-US", {
            style: "currency",
            currency: "USD",
            maximumFractionDigits: 0,
            minimumFractionDigits: 0,
          }).format(bond.purchased)
        )}
      </TableCell>
      <TableCell>
        <Link component={NavLink} to={`/bonds/${bond.name}`}>
          <Button variant="outlined" color="primary">
            <Typography variant="h6">Bond</Typography>
          </Button>
        </Link>
      </TableCell>
    </TableRow>
  );
}
