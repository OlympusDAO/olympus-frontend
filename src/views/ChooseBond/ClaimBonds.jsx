import { useEffect, useState } from "react";
import { t, Trans } from "@lingui/macro";
import { ClaimBondTableData, ClaimBondCardData } from "./ClaimRow";
import { txnButtonText, isPendingTxn, txnButtonTextGeneralPending } from "src/slices/PendingTxnsSlice";
import { redeemAllBonds, redeemBond } from "src/slices/BondSlice";
import { calculateUserBondDetails } from "src/slices/AccountSlice";
import CardHeader from "../../components/CardHeader/CardHeader";
import { useWeb3Context } from "src/hooks/web3Context";
import useBonds from "src/hooks/Bonds";
import {
  Button,
  Box,
  Paper,
  TableContainer,
  TableHead,
  TableBody,
  TableRow,
  TableCell,
  Table,
  Zoom,
} from "@material-ui/core";
import useMediaQuery from "@material-ui/core/useMediaQuery";
import "./choosebond.scss";
import { useSelector, useDispatch } from "react-redux";

function ClaimBonds({ activeBonds }) {
  const dispatch = useDispatch();
  const { provider, address, chainID } = useWeb3Context();
  const { bonds } = useBonds(chainID);

  const [numberOfBonds, setNumberOfBonds] = useState(0);
  const isSmallScreen = useMediaQuery("(max-width: 733px)"); // change to breakpoint query

  const pendingTransactions = useSelector(state => {
    return state.pendingTransactions;
  });

  const pendingClaim = () => {
    if (
      isPendingTxn(pendingTransactions, "redeem_all_bonds") ||
      isPendingTxn(pendingTransactions, "redeem_all_bonds_autostake")
    ) {
      return true;
    }

    return false;
  };

  const onRedeemAll = async ({ autostake }) => {
    console.log("redeeming all bonds");

    await dispatch(redeemAllBonds({ address, bonds, networkID: chainID, provider, autostake }));

    console.log("redeem all complete");
  };

  useEffect(() => {
    let bondCount = Object.keys(activeBonds).length;
    setNumberOfBonds(bondCount);
  }, [activeBonds]);

  return (
    <>
      {numberOfBonds > 0 && (
        <Zoom in={true}>
          <Paper className="ohm-card claim-bonds-card">
            <CardHeader title="Your Bonds (1,1)" />
            <Box>
              {!isSmallScreen && (
                <TableContainer>
                  <Table aria-label="Claimable bonds">
                    <TableHead>
                      <TableRow>
                        <TableCell align="center">
                          <Trans>Bond</Trans>
                        </TableCell>
                        <TableCell align="center">
                          <Trans>Claimable</Trans>
                        </TableCell>
                        <TableCell align="center">
                          <Trans>Pending</Trans>
                        </TableCell>
                        <TableCell align="right">
                          <Trans>Fully Vested</Trans>
                        </TableCell>
                        <TableCell align="right"></TableCell>
                      </TableRow>
                    </TableHead>
                    <TableBody>
                      {Object.entries(activeBonds).map((bond, i) => (
                        <ClaimBondTableData key={i} userBond={bond} />
                      ))}
                    </TableBody>
                  </Table>
                </TableContainer>
              )}

              {isSmallScreen &&
                Object.entries(activeBonds).map((bond, i) => <ClaimBondCardData key={i} userBond={bond} />)}

              <Box
                display="flex"
                justifyContent="center"
                className={`global-claim-buttons ${isSmallScreen ? "small" : ""}`}
              >
                {numberOfBonds > 1 && (
                  <>
                    <Button
                      variant="contained"
                      color="primary"
                      className="transaction-button"
                      fullWidth
                      disabled={pendingClaim()}
                      onClick={() => {
                        onRedeemAll({ autostake: false });
                      }}
                    >
                      {txnButtonTextGeneralPending(pendingTransactions, "redeem_all_bonds", t`Claim all`)}
                    </Button>

                    <Button
                      variant="contained"
                      color="primary"
                      id="claim-all-and-stake-btn"
                      className="transaction-button"
                      fullWidth
                      disabled={pendingClaim()}
                      onClick={() => {
                        onRedeemAll({ autostake: true });
                      }}
                    >
                      {txnButtonTextGeneralPending(
                        pendingTransactions,
                        "redeem_all_bonds_autostake",
                        t`Claim all and Stake`,
                      )}
                    </Button>
                  </>
                )}
              </Box>
            </Box>
          </Paper>
        </Zoom>
      )}
    </>
  );
}

export default ClaimBonds;
