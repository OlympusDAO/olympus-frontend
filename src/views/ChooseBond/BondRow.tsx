import "./ChooseBond.scss";

import { t, Trans } from "@lingui/macro";
import { Button, Link, Paper, Slide, SvgIcon, TableCell, TableRow, Typography } from "@material-ui/core";
import { Skeleton } from "@material-ui/lab";
import { TokenStack } from "@olympusdao/component-library";
import { NavLink } from "react-router-dom";
import { NetworkId } from "src/constants";
import { IAllBondData } from "src/hooks/Bonds";
import { Bond, CustomBond, LPBond } from "src/lib/Bond";

import { ReactComponent as ArrowUp } from "../../assets/icons/arrow-up.svg";
import { useWeb3Context } from "../../hooks";
import { DisplayBondDiscount, DisplayBondPrice } from "../Bond/Bond";

type BondUnion = CustomBond | LPBond;

export function BondDataCard({ bond }: { bond: IAllBondData | Bond }) {
  const { networkId } = useWeb3Context();
  const uBond = bond as BondUnion;
  const allBondData = bond as IAllBondData;
  // Use BondPrice as indicator of loading.
  const isBondLoading = !allBondData?.bondPrice ?? true;

  return (
    <Slide direction="up" in={true}>
      <Paper id={`${bond.name}--bond`} className="bond-data-card ohm-card">
        <div className="bond-pair">
          <TokenStack tokens={bond.bondIconSvg} />
          <div className="bond-name">
            <Typography>{bond.displayName}</Typography>
            {bond.isLP && (
              <div>
                <Link href={uBond?.lpUrl} target="_blank">
                  <Typography variant="body1">
                    <Trans>Deposit LP</Trans>
                    <SvgIcon component={ArrowUp} htmlColor="#A3A3A3" />
                  </Typography>
                </Link>
              </div>
            )}
          </div>
        </div>
        <div className="data-row">
          <Typography>
            <Trans>Price</Trans>
          </Typography>
          <Typography className="bond-price">
            <>
              {isBondLoading ? (
                <Skeleton width="50px" />
              ) : (
                <DisplayBondPrice key={bond.name} bond={allBondData as IAllBondData} />
              )}
            </>
          </Typography>
        </div>
        <div className="data-row">
          <Typography>
            <Trans>ROI</Trans>
          </Typography>
          <Typography>
            {isBondLoading ? (
              <Skeleton width="50px" />
            ) : (
              <DisplayBondDiscount key={bond.name} bond={allBondData as IAllBondData} />
            )}
          </Typography>
        </div>

        <div className="data-row">
          <Typography>
            <Trans>Purchased</Trans>
          </Typography>
          <Typography>
            {isBondLoading ? (
              <Skeleton width="80px" />
            ) : (
              new Intl.NumberFormat("en-US", {
                style: "currency",
                currency: "USD",
                maximumFractionDigits: 0,
                minimumFractionDigits: 0,
              }).format(allBondData?.purchased as number)
            )}
          </Typography>
        </div>
        <Link component={NavLink} to={`/bonds-v1/${bond.name}`}>
          <Button variant="outlined" color="primary" fullWidth disabled={!bond.isBondable[networkId as NetworkId]}>
            <Typography variant="h5">
              {/* NOTE (appleseed): temporary for ONHOLD MIGRATION */}
              {/* {!bond.isBondable[networkId] ? t`Sold Out` : t`Bond ${bond.displayName}`} */}
              {bond.isLOLable[networkId as NetworkId] ? bond.LOLmessage : t`Bond ${bond.displayName}`}
            </Typography>
          </Button>
        </Link>
      </Paper>
    </Slide>
  );
}

export function BondTableData({ bond }: { bond: Bond }) {
  const uBond = bond as BondUnion;
  const allBondData = bond as IAllBondData;
  const { networkId } = useWeb3Context();
  // Use BondPrice as indicator of loading.
  const isBondLoading = !allBondData.bondPrice ?? true;
  // const isBondLoading = useSelector(state => !state.bonding[bond]?.bondPrice ?? true);

  return (
    <TableRow id={`${bond.name}--bond`}>
      <TableCell align="left" className="bond-name-cell">
        <TokenStack tokens={bond.bondIconSvg} />
        <div className="bond-name">
          <Typography variant="body1">{allBondData.displayName}</Typography>
          {bond.isLP && (
            <Link color="primary" href={uBond.lpUrl} target="_blank">
              <Typography variant="body1">
                <Trans>Deposit LP</Trans>
                <SvgIcon component={ArrowUp} htmlColor="#A3A3A3" />
              </Typography>
            </Link>
          )}
        </div>
      </TableCell>
      <TableCell align="left">
        <Typography>
          <>{isBondLoading ? <Skeleton width="50px" /> : <DisplayBondPrice key={bond.name} bond={allBondData} />}</>
        </Typography>
      </TableCell>
      <TableCell align="left">
        {" "}
        {isBondLoading ? <Skeleton width="50px" /> : <DisplayBondDiscount key={bond.name} bond={allBondData} />}
      </TableCell>
      <TableCell align="right">
        {isBondLoading ? (
          <Skeleton />
        ) : (
          new Intl.NumberFormat("en-US", {
            style: "currency",
            currency: "USD",
            maximumFractionDigits: 0,
            minimumFractionDigits: 0,
          }).format(allBondData.purchased)
        )}
      </TableCell>
      <TableCell>
        <Link component={NavLink} to={`/bonds-v1/${bond.name}`}>
          <Button
            variant="outlined"
            color="primary"
            disabled={!bond.isBondable[networkId as NetworkId]}
            style={{ width: "100%" }}
          >
            {/* NOTE (appleseed): temporary for ONHOLD MIGRATION */}
            {/* <Typography variant="h6">{!bond.isBondable[networkId] ? t`Sold Out` : t`do_bond`}</Typography> */}
            <Typography variant="h6">
              {bond.isLOLable[networkId as NetworkId] ? bond.LOLmessage : t`do_bond`}
            </Typography>
          </Button>
        </Link>
      </TableCell>
    </TableRow>
  );
}
