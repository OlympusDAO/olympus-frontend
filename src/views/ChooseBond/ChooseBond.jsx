import React, { useEffect, useState } from "react";
import { useSelector } from "react-redux";
import { Grid, Table, TableBody, TableCell, TableContainer, TableHead, TableRow } from "@material-ui/core";
import { Card } from "rimble-ui";
import "../Stake/stake.scss";
import { BondTableData, BondCardData } from "./BondRow";
import { BONDS } from "../../constants";
import useMediaQuery from "@material-ui/core/useMediaQuery";
import { trim } from "../../helpers";
import useBonds from "../../hooks/Bonds";

function ChooseBond() {
  const marketPrice = useSelector(state => {
    return state.app.marketPrice;
  });

  const isSmallScreen = useMediaQuery("(max-width: 1125px)");
  const isMediumScreen = useMediaQuery("(min-width: 1279px, max-width: 1500px)");
  const isVerySmallScreen = useMediaQuery("(max-width: 589px)");

  const treasuryBalance = useSelector(state => {
    return state.app.treasuryBalance;
  });

  const bonds = useBonds();

  return (
    <Grid container id="choose-bond-view" justify="center" spacing={2}>
      <Card className={`ohm-card secondary ${isSmallScreen && "mobile"} ${isMediumScreen && "med"}`}>
        <div className="card-content">
          <Grid container item xs={12} spacing={2}>
            <Grid item sm={7} lg={9}>
              <h3>Treasury Balance</h3>
              <h2 className="content">
                {treasuryBalance &&
                  new Intl.NumberFormat("en-US", {
                    style: "currency",
                    currency: "USD",
                    maximumFractionDigits: 0,
                    minimumFractionDigits: 0,
                  }).format(treasuryBalance)}
              </h2>
            </Grid>

            <Grid item xs={5} sm={5} lg={3} className={`ohm-price ${isVerySmallScreen && "very-small"}`}>
              <h3>OHM Price</h3>
              <h2 className="content">{trim(marketPrice, 2)}</h2>
            </Grid>
          </Grid>
        </div>
      </Card>

      <Card className={`ohm-card primary ${isSmallScreen && "mobile"} ${isMediumScreen && "med"}`}>
        <div className="card-header" style={{ background: "transparent" }}>
          <h5>Bonds (1, 1)</h5>
        </div>
        {!isSmallScreen ? (
          <div className="card-content">
            <TableContainer>
              <Table aria-label="Available bonds">
                <TableHead>
                  <TableRow>
                    <TableCell align="left">Bond</TableCell>
                    <TableCell align="center">Price</TableCell>
                    <TableCell>ROI</TableCell>
                    <TableCell>Purchased</TableCell>
                    <TableCell align="right"></TableCell>
                  </TableRow>
                </TableHead>
                <TableBody>
                  {bonds.map(bond => (
                    <BondTableData key={bond.value} bond={bond.value} />
                  ))}
                </TableBody>
              </Table>
            </TableContainer>
          </div>
        ) : (
          <>
            {/* { Object.keys(BONDS).map(bond => ( */}
            {[BONDS.ohm_dai, BONDS.dai, BONDS.ohm_frax, BONDS.frax].map(bond => (
              <div className="card-content" key={bond}>
                <BondCardData key={bond} bond={bond} />
              </div>
            ))}
          </>
        )}
      </Card>
    </Grid>
  );
}

export default ChooseBond;
