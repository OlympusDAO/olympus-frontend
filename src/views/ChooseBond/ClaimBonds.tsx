import "./ChooseBond.scss";

import { Trans } from "@lingui/macro";
import { Box, Table, TableBody, TableCell, TableContainer, TableHead, TableRow } from "@material-ui/core";
import useMediaQuery from "@material-ui/core/useMediaQuery";
import { useEffect, useState } from "react";
import { IUserBondDetails } from "src/slices/AccountSlice";

import { ClaimBondCardData, ClaimBondTableData } from "./ClaimRow";

export function ClaimBondsSubComponent({ activeBonds }: { activeBonds: IUserBondDetails[] }) {
  const [, setNumberOfBonds] = useState(0);
  const isSmallScreen = useMediaQuery("(max-width: 733px)"); // change to breakpoint query

  useEffect(() => {
    const bondCount = activeBonds.length;
    setNumberOfBonds(bondCount);
  }, [activeBonds]);

  return (
    <Box style={{ width: "100%" }}>
      {!isSmallScreen && (
        <TableContainer>
          <Table aria-label="Claimable bonds">
            <TableHead>
              <TableRow>
                <TableCell align="center">
                  <Trans>Bond</Trans>
                </TableCell>
                <TableCell align="center">
                  <Trans>Claimable</Trans>
                </TableCell>
                <TableCell align="center">
                  <Trans>Pending</Trans>
                </TableCell>
                <TableCell align="right">
                  <Trans>Fully Vested</Trans>
                </TableCell>
                <TableCell align="right"></TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {Object.entries(activeBonds).map((bond, i) => (
                <ClaimBondTableData key={i} userBond={bond} />
              ))}
            </TableBody>
          </Table>
        </TableContainer>
      )}

      {isSmallScreen && Object.entries(activeBonds).map((bond, i) => <ClaimBondCardData key={i} userBond={bond} />)}
    </Box>
  );
}
