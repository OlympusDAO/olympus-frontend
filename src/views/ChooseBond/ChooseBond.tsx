import "./ChooseBond.scss";

import { t, Trans } from "@lingui/macro";
import {
  Box,
  ButtonBase,
  Grid,
  SvgIcon,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableRow,
  Typography,
  Zoom,
} from "@material-ui/core";
import useMediaQuery from "@material-ui/core/useMediaQuery";
import { Metric, MetricCollection, Paper } from "@olympusdao/component-library";
import isEmpty from "lodash/isEmpty";
import { useHistory } from "react-router";
import { Link } from "react-router-dom";
import { allBondsMap } from "src/helpers/AllBonds";
import { useAppSelector } from "src/hooks";
import { usePathForNetwork } from "src/hooks/usePathForNetwork";
import { useWeb3Context } from "src/hooks/web3Context";
import { IUserBondDetails } from "src/slices/AccountSlice";
import { IUserNote } from "src/slices/BondSliceV2";

import { ReactComponent as ArrowUp } from "../../assets/icons/arrow-up.svg";
import { formatCurrency } from "../../helpers";
import useBonds from "../../hooks/Bonds";
import ClaimBonds from "../BondV2/ClaimBonds";
import { BondDataCard, BondTableData } from "./BondRow";

function ChooseBond() {
  const { networkId } = useWeb3Context();
  const history = useHistory();
  const { bonds } = useBonds(networkId);
  usePathForNetwork({ pathName: "bonds", networkID: networkId, history });
  const isSmallScreen = useMediaQuery("(max-width: 733px)"); // change to breakpoint query
  const isVerySmallScreen = useMediaQuery("(max-width: 420px)");

  const isAppLoading: boolean = useAppSelector(state => state.app.loading);
  const isAccountLoading: boolean = useAppSelector(state => state.account.loading);

  const accountNotes: IUserNote[] = useAppSelector(state => state.bondingV2.notes);

  const accountBonds: IUserBondDetails[] = useAppSelector(state => {
    const withInterestDue = [];
    for (const bond in state.account.bonds) {
      if (state.account.bonds[bond].interestDue > 0) {
        withInterestDue.push(state.account.bonds[bond]);
      }
    }
    return withInterestDue;
  });

  const marketPrice: number | undefined = useAppSelector(state => {
    return state.app.marketPrice;
  });

  const treasuryBalance: number | undefined = useAppSelector(state => {
    if (state.bonding.loading == false) {
      let tokenBalances = 0;
      for (const bond in allBondsMap) {
        if (state.bonding[bond]) {
          tokenBalances += state.bonding[bond].purchased;
        }
      }
      return tokenBalances;
    }
  });

  const formattedTreasuryBalance = new Intl.NumberFormat("en-US", {
    style: "currency",
    currency: "USD",
    maximumFractionDigits: 0,
    minimumFractionDigits: 0,
  }).format(Number(treasuryBalance));

  const topRightCTA = (
    <ButtonBase>
      <Typography style={{ lineHeight: "33px" }}>
        <b>
          <Link to="/bonds" style={{ textDecoration: "none", color: "inherit" }}>
            <Trans>v2 bonds</Trans>
            <SvgIcon
              style={{ margin: "0 0 0 5px", verticalAlign: "text-bottom" }}
              component={ArrowUp}
              color="primary"
            />
          </Link>
        </b>
      </Typography>
    </ButtonBase>
  );
  return (
    <div id="choose-bond-view">
      {(!isEmpty(accountNotes) || !isEmpty(accountBonds)) && <ClaimBonds activeNotes={accountNotes} />}
      <Zoom in={true}>
        <Paper headerText={`${t`Bond`} (1,1)`} topRight={topRightCTA}>
          <MetricCollection>
            <Metric
              label={t`Treasury Balance`}
              metric={formattedTreasuryBalance}
              isLoading={!!treasuryBalance ? false : true}
            />
            <Metric
              label={t`OHM Price`}
              metric={formatCurrency(Number(marketPrice), 2)}
              isLoading={marketPrice ? false : true}
            />
          </MetricCollection>

          {!isSmallScreen && (
            <Grid container item>
              <TableContainer>
                <Table aria-label="Available bonds">
                  <TableHead>
                    <TableRow>
                      <TableCell align="center">
                        <Trans>Bond</Trans>
                      </TableCell>
                      <TableCell align="left">
                        <Trans>Price</Trans>
                      </TableCell>
                      <TableCell align="left">
                        <Trans>ROI</Trans>
                      </TableCell>
                      <TableCell align="right">
                        <Trans>Purchased</Trans>
                      </TableCell>
                      <TableCell align="right"></TableCell>
                    </TableRow>
                  </TableHead>
                  <TableBody>
                    {bonds.map(bond => {
                      // NOTE (appleseed): temporary for ONHOLD MIGRATION
                      // if (bond.getBondability(networkId)) {
                      if (bond.getBondability(networkId) || bond.getLOLability(networkId)) {
                        return <BondTableData key={bond.name} bond={bond} />;
                      }
                    })}
                  </TableBody>
                </Table>
              </TableContainer>
            </Grid>
          )}
        </Paper>
      </Zoom>

      {isSmallScreen && (
        <Box className="ohm-card-container">
          <Grid container item spacing={2}>
            {bonds.map(bond => {
              // NOTE (appleseed): temporary for ONHOLD MIGRATION
              // if (bond.getBondability(networkId)) {
              if (bond.getBondability(networkId) || bond.getLOLability(networkId)) {
                return (
                  <Grid item xs={12} key={bond.name}>
                    <BondDataCard key={bond.name} bond={bond} />
                  </Grid>
                );
              }
            })}
          </Grid>
        </Box>
      )}
    </div>
  );
}

export default ChooseBond;
