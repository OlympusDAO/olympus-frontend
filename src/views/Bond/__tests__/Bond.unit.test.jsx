import { render } from "src/testUtils";

import ChooseBond from "../../ChooseBond/ChooseBond";

describe("<ChooseBond/>", () => {
  it("should render component", () => {
    const { container } = render(<ChooseBond />);
    expect(container).toMatchSnapshot();
  });
});
