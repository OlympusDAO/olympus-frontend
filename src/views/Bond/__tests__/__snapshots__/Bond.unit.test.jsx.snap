// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChooseBond/> should render component 1`] = `
<div>
  <div
    id="choose-bond-view"
  >
    <div
      class="MuiPaper-root makeStyles-root-1 makeStyles-root-2  MuiPaper-elevation0 MuiPaper-rounded"
      style="transform: none; webkit-transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column"
      >
        <div
          class="MuiGrid-root card-header MuiGrid-item"
        >
          <div
            class="MuiBox-root MuiBox-root-3"
          >
            <div
              class="MuiBox-root MuiBox-root-4"
            >
              <h5
                class="MuiTypography-root header-text MuiTypography-h5"
              >
                Bond (1,1)
              </h5>
            </div>
            <div
              class="top-right"
            >
              <button
                class="MuiButtonBase-root"
                tabindex="0"
                type="button"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1"
                  style="line-height: 33px;"
                >
                  <b>
                    <a
                      href="/bonds"
                      style="text-decoration: none;"
                    >
                      v2 bonds
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall"
                        focusable="false"
                        style="margin: 0px 0px 0px 5px; vertical-align: text-bottom;"
                        viewBox="0 0 20 20"
                      >
                        arrow-up.svg
                      </svg>
                    </a>
                  </b>
                </p>
              </button>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiBox-root MuiBox-root-5"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-flex-end"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
              >
                <div
                  class="makeStyles-root-6 "
                >
                  <div
                    class="MuiBox-root MuiBox-root-7"
                  >
                    <h5
                      class="MuiTypography-root MuiTypography-h5 MuiTypography-colorTextSecondary"
                    >
                      Treasury Balance
                    </h5>
                    <h4
                      class="MuiTypography-root MuiTypography-h4"
                      style="width: 100%;"
                    >
                      <span
                        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                        style="width: 100%;"
                      />
                    </h4>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
              >
                <div
                  class="makeStyles-root-6 "
                >
                  <div
                    class="MuiBox-root MuiBox-root-8"
                  >
                    <h5
                      class="MuiTypography-root MuiTypography-h5 MuiTypography-colorTextSecondary"
                    >
                      OHM Price
                    </h5>
                    <h4
                      class="MuiTypography-root MuiTypography-h4"
                      style="width: 100%;"
                    >
                      <span
                        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                        style="width: 100%;"
                      />
                    </h4>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item"
          >
            <div
              class="MuiTableContainer-root"
            >
              <table
                aria-label="Available bonds"
                class="MuiTable-root"
              >
                <thead
                  class="MuiTableHead-root"
                >
                  <tr
                    class="MuiTableRow-root MuiTableRow-head"
                  >
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                      scope="col"
                    >
                      Bond
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
                      scope="col"
                    >
                      Price
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
                      scope="col"
                    >
                      Discount
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                      scope="col"
                    >
                      Purchased
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                      scope="col"
                    />
                  </tr>
                </thead>
                <tbody
                  class="MuiTableBody-root"
                />
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
